controller/lib/core/sensors.cpp

```diff
@@ -123,22 +117,16 @@ Pressure Sensors::ReadPressureSensor(Sensor s) {
 }
 
 /*static*/ VolumetricFlow Sensors::PressureDeltaToFlow(Pressure delta) {
-  // TODO(jlebar): Make these constexpr once we have a C++ standard library
-  // PortArea must be larger than the ChokeArea [meters^2]
-  float venturiPortArea = diameter_to_area_m2(DEFAULT_VENTURI_PORT_DIAM);
-  float venturiChokeArea = diameter_to_area_m2(DEFAULT_VENTURI_CHOKE_DIAM);
-  //[meters^4]
-  float venturiAreaProduct = venturiPortArea * venturiChokeArea;
-  // Equivalent to 1/sqrt(A1^2 - A2^2) guaranteed never to have a negative
-  // radicand [1/meters^2]
-  float bernoulliAreaDivisor =
-      1.0f / sqrtf(venturiPortArea * venturiPortArea -
-                   venturiChokeArea * venturiChokeArea);
-
-  float sgn = copysignf(1.0f, delta.kPa());
-  return cubic_m_per_sec(sgn * sqrtf(2 / DENSITY_OF_AIR_KG_PER_CUBIC_METER) *
-                         venturiAreaProduct * bernoulliAreaDivisor *
-                         sqrtf(abs(delta.kPa()) * 1000.0f));
+  auto pow2 = [](float x) { return x * x; };
+  auto diameter_to_area_m2 = [](Length d) {
+    return static_cast<float>(M_PI) / 4.0f * d.meters() * d.meters();
+  };
+  constexpr float portArea = diameter_to_area_m2(DEFAULT_VENTURI_PORT_DIAM);
```

Comment:
Cool that this works, I guess lambdas are constexpr by default?

Reply:
Yes in C++17, https://en.cppreference.com/w/cpp/language/lambda